### DISCUSSION TITLE
Public Roadmap

### DISCUSSION THREAD

**Discussion by e-p-s** [TriliumNext/Notes#464] [General]
Hi team,

First of all, I want to express my appreciation for all the hard work you’ve put into this project. I’ve been following along and really enjoy the direction it’s headed.

However, as a user and contributor, I find it difficult to stay up-to-date with ongoing development and future plans. It’s often unclear what features or improvements are currently being worked on or planned. To help with this, I’d like to propose the creation of a public roadmap, similar to [Linkwarden's roadmap](https://github.com/orgs/linkwarden/projects/1).

A public roadmap would provide a clear overview of:

Current features in development
Upcoming ideas or improvements
Prioritized bug fixes
This could also encourage more community involvement, as users can better understand the direction of the project and even contribute by providing feedback or helping with tasks currently in development.

Here are a few benefits of a public roadmap:

Transparency: Contributors and users can see the project’s progress and future goals.
Community Engagement: Users could offer feedback on planned features or submit their own suggestions.
Clearer Priorities: Helps the community understand which features or fixes are prioritized.

This is just a suggestion to improve the overall project experience, and I believe it would benefit both the community and maintainers. Thank you for considering this idea!

Looking forward to your thoughts.

Best regards,
Chris

**Comment by eliandoran** [comment]
Hello @e-p-s ,,

I appreciate you taking the time to propose this.

For the time being I don't think we achieve a clear road map. The project is currently developed by only a handful of people (see https://github.com/TriliumNext/Notes/graphs/contributors?from=4%2F27%2F2024) with help on the organization side from people like @meichthys . There is no long-term plan that can be shared because the development is done in our spare time with no time commitment as of now.

In general, every developer is free to work on whatever they wish as long as it's in the direction we want to take it as a group (which is also a bit fuzzy). The main topics I would personally address are:

- An improved code base (for example the migration to TypeScript).
- All libraries and dependencies kept up to date.
- Internationalization support (a lot of progress was done on this front)
- Improved mobile experience.
- Most importantly for me would be to improve the user experience: a "better" default theme, reducing visual clutter in the layout, etc.

As the team will grow bigger and more organized we can of course consider such a roadmap.

**Comment by EclipseNow** [comment]
> Improved mobile experience.

Hi,
first - thank you developers!

Second - does this mean an app?

**Reply by meichthys** [reply]
https://github.com/TriliumNext/Notes/issues/72
